lib/puppet/type/resources.rb

```diff
@@ -120,13 +120,7 @@ def generate
       select { |r| r.class.validproperty?(:ensure) }.
       select { |r| able_to_ensure_absent?(r) }.
       each { |resource|
-        @parameters.each do |name, param|
-          resource[name] = param.value if param.metaparam?
-        end
-
-        # Mark that we're purging, so transactions can handle relationships
-        # correctly
-        resource.purging
+        resource.copy_metaparams(@parameters)
```

Comment:
Do we still need to call `resource_purging` after copying metaparams?

Reply:
Good catch, re-added it in